parser/parser_test.go

```diff
@@ -12,35 +10,50 @@ var repoMap = map[string]string{
 	"lg2": "libgit2/libgit2",
 }
 
-func testExpandRepo(t *testing.T, input string, repo string) {
-	result := Parse(repoMap, input)
-	msg := fmt.Sprintf("input %#v expected repo %#v to be %#v", input, result.Repo, repo)
-	assert.Equal(t, repo, result.Repo, msg)
-}
+func TestParse(t *testing.T) {
+	repoTests := []struct {
+		input string // the input
+		repo  string // the expected repo match or expansion
+	}{
+		{"", ""},                     // no input, no repo match
+		{"df", "zerowidth/dotfiles"}, // match shorthand
+		{" df", ""},                  // no match, leading space
+		{"foo/bar", "foo/bar"},       // fully qualified
+	}
 
-func testExpandRepoIssue(t *testing.T, input string, repo string, issue string) {
-	result := Parse(repoMap, input)
-	msg := fmt.Sprintf("input %#v expected repo %#v to be %#v", input, result.Repo, repo)
-	assert.Equal(t, repo, result.Repo, msg)
-	msg = fmt.Sprintf("input %#v expected issue %#v to be %#v", input, result.Issue, issue)
-	assert.Equal(t, issue, result.Issue, msg)
-}
+	repoIssueTests := []struct {
+		input string // the input
+		repo  string // the expected repo match or expansion
+		issue string // the expected issue match
+	}{
+		{"", "", ""}, // no issue nor repo
+		{"df 123", "zerowidth/dotfiles", "123"},
+		{"df#123", "zerowidth/dotfiles", "123"},
+		{"df #123", "zerowidth/dotfiles", "123"}, // space and hash
+		{"df123", "zerowidth/dotfiles", "123"},   // prefix match
+		{"df2 34", "zerowidth/dotfiles2", "34"},
+		{"df234", "zerowidth/dotfiles2", "34"}, // prefix match
+		{"lg2 123", "libgit2/libgit2", "123"},
+		{"lg2123", "libgit2/libgit2", "123"},  // prefix match
+		{"foo/bar 123", "foo/bar", "123"},     // fully qualified
+		{"df 0123", "zerowidth/dotfiles", ""}, // invalid issue
+	}
 
-func TestParse(t *testing.T) {
-	testExpandRepo(t, "", "")
-	testExpandRepo(t, "df", "zerowidth/dotfiles") // match shorthand
-	testExpandRepo(t, " df", "")                  // no match, leading space
-	testExpandRepo(t, "foo/bar", "foo/bar")       // fully qualified
+	for _, tc := range repoTests {
```

Comment:
I'd extract the assertion block into something with the signature `func(t *testing.T)` so you can use `t.Run`, which treat each run as a subtest (see https://golang.org/pkg/testing/#T.Run).